src/sas/sascalc/fit/expression.py

```diff
@@ -379,10 +377,10 @@ def order_dependencies(pairs):
         else:
             left, right = [set(s) for s in zip(*pairs)]
             resolved = dependent - left
-        #print "independent", independent, "dependent", dependent, "resolvable", resolved
+
         order += resolved
-        #print "new order", order
-    order.reverse()
+
+    #order.reverse()
```

Comment:
That doesn't seem right. I'm pulling the independent variables from the list until there are no more independent variables, so the last variables pulled must be the first ones evaluated. That's why the order is reversed.

Removing the `reverse()` breaks the tests in `expression.py`(not that you can tell because pytest isn't run on the source tree, only the test tree).

To debug this you are going to need to print the function produced in `expression._compile_constraints` to make sure it is what you are expecting.

Reply:
I just checked this empirically, simply verifying the output of order_dependencies (and the result in SasView), and I see that the order of the dependencies is well determined before doing the reverse. For example, assuming that there is a single independent parameter (M1), but that a user will define constraints as M2=M1, M3=M2, and M4=M3, the function (without the reverse) returns the expected order [M2, M3, M4]. So I did not go yet in the detail of the logic of the function, but I'm sure that at least in my test case the reverse will make things fail.